File: 01_Array/Q05_169. Majority Element/solution.cpp
// Problem Link -> https://leetcode.com/problems/majority-element/description/

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, int> freq;

        for(auto &it : nums){
            freq[it]++;
        }

        for(auto &element : freq){
            if(element.second > n/2) return element.first;
        }

        return -1;
    }
};



// Time Complexity -> O(n)
// Space Complexity -> O(n)

File: 01_Array/Q05_169. Majority Element/solution.py
# Problem Link -> https://leetcode.com/problems/majority-element/description/

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        freq = {}

        for num in nums:
            if num in freq:
                freq[num] += 1
            else:
                freq[num] = 1
        
        for key, value in freq.items():
            if value > n/2:
                return key
        
        return -1


      
# Time Complexity -> O(n)
# Space Complexity -> O(n)
